### DISCUSSION TITLE
PC Engine ROM CRC checking issues

### DISCUSSION THREAD

**Discussion by PsyK0p4T** [sanni/cartreader#329] [Coding Corner]
Hello people

I finally took the time to build a PCE to SNES slot adapter
![adapter](https://user-images.githubusercontent.com/87064902/154583905-d0cc75fb-5456-48cc-a1bd-4c8228bb9446.jpg)

Then tried to dump a HuCard with it : "Naxat Open".
Dumping process ran perfectly fine and I got the expected CRC for it "60ECAE22". So no problem with dumping.
But then the CRC is checked and compared with SD txt file entries. Here comes the issue :

![display](https://user-images.githubusercontent.com/87064902/154583934-42a7d54f-fd55-4a31-85a8-e5cd8932a177.jpg)

The just dumped ROM is opened, its CRC is calculated (and displayed) fine, but then something goes wrong with compare process.
I'm not really at ease with FS but something is maybe failing around this point : 
get_line(gamename, &script, 96);
get_line(crc_search, &script, 9);

gamename var seems to store more than a line of the txt, and crc_search var looks empty when I try to display them on screen. 
So... i guess the strcmp function always fails, and it leads to the Chksum Error message whereas I got the expected one.
If someone can investigate... it would be cool ;)

**Comment by tamanegitaro** [comment]
Maybe because the CRC file has wrong new line code. Try to convert new line code to different one.

Good luck.
tamanegi_taro

**Reply by PsyK0p4T** [reply]
The CRC is correct in the txt file (just checked again), and the CRC obtained during dump is correct too.
_Naxat Open
60ECAE22_
That's why I'm saying there is maybe a problem during txt read, or string compare operation.

get_line(gamename, &script, 96);
"gamename" var seems to store more chars than expected and "crc_search" looks empty.
Can you try it from your side ?

**Reply by tamanegitaro** [reply]
Yep. Your understanding is correct. Both CRCs in txt and calculated are same.
But “new line code” used in the text file is different from expected value. 



unix is LF “\n”.
Mac is CR “\r”
Windows is CR+LF “\r\n”

So you need to convert new line code of txt file to different one. Did you get what I mean?
If you still didn’t get what I mean, I’ll get you correct txt file this weekend.

Good luck,
tamanegi_taro
